ksp_plugin_adapter/differential_slider.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Globalization;
+
+namespace principia {
+namespace ksp_plugin_adapter {
+
+internal class DifferentialSlider {
+  private float slider_position_ = 0.0f;
+  private DateTime last_time_;
+
+  private readonly string label_;
+  private readonly string unit_;
+  private readonly CultureInfo culture_;
+
+  private readonly double log10_lower_rate_ = -3;
+  private readonly double log10_upper_rate_ = 3.5;
+  private readonly double min_value_;
+  private readonly double max_value_;
+
+  public double value { get; set; }
+
+  public DifferentialSlider(string label,
+                            string unit,
+                            double log10_lower_rate,
+                            double log10_upper_rate)
+      : this(label,
+             unit,
+             log10_lower_rate,
+             log10_upper_rate,
+             -double.NegativeInfinity,
+             -double.PositiveInfinity) {}
+
+  // Rates are in unite of |value| per real-time second.
+  public DifferentialSlider(string label,
+                            string unit,
+                            double log10_lower_rate,
+                            double log10_upper_rate,
+                            double min_value,
+                            double max_value) {
+    label_ = label;
+    unit_ = unit;
+    culture_= new CultureInfo("");
+    culture_.NumberFormat.NumberGroupSeparator = "'";
+    log10_lower_rate_ = log10_lower_rate;
+    log10_upper_rate_ = log10_upper_rate;
+    min_value_ = min_value;
+    max_value_ = max_value;
+  }
+
+  // Returns true if and only if |value| changed.
+  public bool Render() {
+    UnityEngine.GUI.skin = null;
+    UnityEngine.GUILayout.BeginHorizontal();
+
+    UnityEngine.GUILayout.Label(
+        text       : label_,
+        options    : UnityEngine.GUILayout.Width(100));
```

Comment:
Could we have constants for 100 (here and at line 63), 50 (line 67), 21 (lines 71-72) and 20 (line 80)?